packages/cactus-plugin-ledger-connector-iroha/src/test/typescript/integration/run-transaction-endpoint-v1.test.ts

```diff
@@ -0,0 +1,783 @@
+import http from "http";
+import { AddressInfo } from "net";
+import test, { Test } from "tape-promise/tape";
+import { v4 as uuidv4 } from "uuid";
+import bodyParser from "body-parser";
+import express from "express";
+//import Promise from "bluebird";
+import {
+  Containers,
+  //IrohaTestLedger,
+  pruneDockerAllIfGithubAction,
+  PostgresTestContainer,
+  IrohaTestLedger,
+} from "@hyperledger/cactus-test-tooling";
+import { PluginRegistry } from "@hyperledger/cactus-core";
+import { PluginImportType } from "@hyperledger/cactus-core-api";
+
+import {
+  IListenOptions,
+  LogLevelDesc,
+  Servers,
+} from "@hyperledger/cactus-common";
+
+import {
+  PluginLedgerConnectorIroha,
+  DefaultApi as IrohaApi,
+  //RunTransactionRequest,
+  PluginFactoryLedgerConnector,
+  RunTransactionRequest,
+} from "../../../main/typescript/public-api";
+
+import { Configuration } from "@hyperledger/cactus-core-api";
+
+const testCase = "runs tx on an Iroha v1.2.0 ledger";
+const logLevel: LogLevelDesc = "TRACE";
+
+test.onFailure(async () => {
+  await Containers.logDiagnostics({ logLevel });
+});
+
+test("BEFORE " + testCase, async (t: Test) => {
+  const pruning = pruneDockerAllIfGithubAction({ logLevel });
+  await t.doesNotReject(pruning, "Pruning didn't throw OK");
+  t.end();
+});
+
+test(testCase, async (t: Test) => {
+  const postgres = new PostgresTestContainer({
+    containerImageName: "postgres",
+    containerImageVersion: "9.5-alpine",
+    postgresPort: 5432,
+    envVars: ["POSTGRES_USER=postgres", "POSTGRES_PASSWORD=mysecretpassword"],
+  });
+
+  const iroha = new IrohaTestLedger({
+    containerImageVersion: "1.20c",
+    containerImageName: "hanxyz/iroha",
```

Comment:
@hxlaf We need to Publish the container image on the official repository as well and then update the code here to point to that one instead.
This is something that I need to do so please bug me about publishing the image and then I'll be able to provide the new tag for you to change it in the code.

Reply:
Thank you! I already gzipped my image via docker save. Then I uploaded it to [my Github](https://github.com/hxlaf/iroha_docker/blob/dockerImage/iroha_1_20c.tar.gz) so that you may easily download.
Kindly load my image with:
docker load <docker image name>.tar.gz
[](https://docs.docker.com/engine/reference/commandline/load/)
Let me know if you need anything else.

Reply:
Resolved!